sycl/include/CL/sycl/detail/queue_impl.hpp

```diff
@@ -19,154 +19,218 @@
 #include <CL/sycl/exception_list.hpp>
 #include <CL/sycl/handler.hpp>
 #include <CL/sycl/property_list.hpp>
+#include <CL/sycl/stl.hpp>
 
 __SYCL_INLINE namespace cl {
 namespace sycl {
 namespace detail {
 
-using ContextImplPtr = std::shared_ptr<detail::context_impl>;
-using DeviceImplPtr = std::shared_ptr<detail::device_impl>;
+using ContextImplPtr = shared_ptr_class<detail::context_impl>;
+using DeviceImplPtr = shared_ptr_class<detail::device_impl>;
 
-// Set max number of queues supported by FPGA RT.
+/// Sets max number of queues supported by FPGA RT.
 const size_t MaxNumQueues = 256;
 
 enum QueueOrder { Ordered, OOO };
 
 class queue_impl {
 public:
+  /// Constructs a SYCL queue from a device using an async_handler and
+  /// property_list provided.
+  ///
+  /// @param Device is a pointer to device_impl.
+  /// @param AsyncHandler is a SYCL asynchronous exception handler.
+  /// @param Order specifies whether the queue being constructed as in-order or
+  /// out-of-order.
+  /// @param PropList is a list of properties to use for queue construction.
   queue_impl(DeviceImplPtr Device, async_handler AsyncHandler,
              QueueOrder Order, const property_list &PropList)
       : queue_impl(Device,
                    detail::getSyclObjImpl(
                        context(createSyclObjFromImpl<device>(Device))),
                    AsyncHandler, Order, PropList){};
 
+  /// Constructs a SYCL queue with an async_handler and property_list provided
+  /// form a device and a context.
+  ///
+  /// @param Device is a pointer to device_impl.
```

Comment:
Do you see any value in the comments like this?
This comment repeats information, which I can get form the constructor declaration - I see the type of `Device` parameter.
You could say something useful about relations between the queue and device - e.g. that commands submitted to the queue will be executed on the `Device`. 
I would expect a comment clarifying why `Device` is needed to construct the queue, not what is the parameter type.

I really appreciate your work with commenting the code, but please, do not add comments just for the sake of having some comments.